packages/prettier-plugin/test/numbers.test.ts

```diff
@@ -0,0 +1,45 @@
+import { format } from "./helpers.js";
+
+// In general, this behavior matches prettier for JS.
+describe("numbers", () => {
+  test("integer", async () => {
+    expect(await format("123")).toBe("123");
+  });
+
+  test("negative integer", async () => {
+    expect(await format("-123")).toBe("-123");
+  });
+
+  test("fixed point", async () => {
+    expect(await format("1.5")).toBe("1.5");
+  });
+
+  test("fixed point with trailing zero", async () => {
+    expect(await format("1.0")).toBe("1.0");
+  });
+
+  test("fixed point with multiple trailing zeroes", async () => {
+    expect(await format("1.000")).toBe("1.0");
+  });
+
+  test("fixed point without fractional part", async () => {
+    expect(await format("1.")).toBe("1");
+  });
+
+  test("floating point with fake fractional part", async () => {
+    expect(await format("1.e3")).toBe("1e3");
+  });
+  test("floating point with single zero fractional part", async () => {
+    expect(await format("1.0e3")).toBe("1.0e3");
+  });
+  test("floating point with multiple zeroes as fractional part", async () => {
+    expect(await format("1.000e3")).toBe("1.0e3");
+  });
+
+  test("floating point with explicit plus sign for exponent", async () => {
+    expect(await format("1e+3")).toBe("1e3");
+  });
+  test("floating point with negative exponent", async () => {
+    expect(await format("1e-3")).toBe("1e-3");
+  });
```

Comment:
All these tests follow the same patern/code, so I probably would have done them in a separate data structure.

Like,
```js
[ 
  ["integer", "123", "123"],
   ...
 ]
``

Reply:
I do like the comprehensiveness of these tests.

Reply:
I've refactored this in #2038.

Reply:
I agree, but then I should refactor all prettier tests (they all follow this pattern), I guess [test.each](https://jestjs.io/docs/api#testeachtablename-fn-timeout) is the best fit here.